### DISCUSSION TITLE
ConsumerWidget as replacement for StatelessWidget

### DISCUSSION THREAD

**Discussion by emailvenkatm** [rrousselGit/riverpod#1441] [Q&A]
Hello,

I am new to flutter development. Very sorry if this is a dumb question.

 In my understanding stalesswidget is something that doesn’t depend on state and never rebuilds even after there is a change in the state. But as per Riverpod documentation consumerwidget is a replacement for statelesswidget. Consumerwidget rebuilds if the state it is subscribed to gets changed right(through ref.watch). Ideally as per flutter documentation statelesswidget shouldn’t get changed because of state changes correct? So how a consumerwidget is a replacement for statelesswidget is my confusion 🙂

**Comment by rrousselGit** [comment]
> In my understanding stalesswidget is something that doesn’t depend on state and never rebuilds even after there is a change in the state.

That isn't the case

StatelessWidget can rebuild in many occasions. For example if using MediaQuery.of and the screen size changes


StatelessWidget only means that it's written in a single immutable class

**Reply by emailvenkat** [reply]
Thanks for the reply @rrousselGit. I just read the documentation too. Thanks for the explaination

**Comment by emailvenkatm** [comment]
One more followup question. We should be we global scope providers vs the local scope for a specific widget. And when is it necessary to use ConfumerStatefulWidget because now even ConsumerWidget can respond to state changes.

**Reply by tbm98** [reply]
`ConsumerStatefulWidget` help you able to use `ref` in `initState`
